client/trino-client/src/main/java/io/trino/client/ResultRows.java

```diff
@@ -74,6 +76,8 @@ public String toString()
     static ResultRows wrapIterator(CloseableIterator<List<Object>> iterator)
     {
         return new ResultRows() {
+            private final AtomicBoolean fetched = new AtomicBoolean();
```

Comment:
nit: make it just `volatile boolean`. It is not safe for concurrent access anyway